File: Data Structures & Algorithms/maximum-subarray/submission-1.py

class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        ans, curSum = nums[0], 0
        for num in nums:
            if curSum < 0:
                curSum = 0
            curSum += num
            ans = max(ans, curSum)
        return ans

File: Data Structures & Algorithms/house-robber/submission-1.js

class Solution {
		rob(nums) {
		let ans = 0;
		function dfs(i) {
			if (i >= nums.length) return 0;
			return  Math.max(dfs(i + 1), nums[i]+ dfs(i + 2));

		}
		return dfs(0);
		

	}



	

}
